### PR TITLE
docs: update developing

### PR DESCRIPTION
**Pull request by wischli**
# Description

Updates the developer docs by removing some deprecated bits. All lines were tested locally. Used as base for outstanding _How to contribute_ docs in docs repository.

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/DEVELOPING.md [hunk 1]
```diff
@@ -84,33 +84,28 @@ You can play with it from the parachain client, make transfers, inspect events,
 ## Linting
 
 ### Source code
-Lint the source code with `cargo +nightly fmt`. This excludes certain paths (defined in `rustfmt.toml`) that we want to stay as close as possible to `paritytech/substrate` to simplify upgrading to new releases.
+Lint the source code with `cargo fmt --all`. This excludes certain paths (defined in `rustfmt.toml`) that we want to stay as close as possible to `paritytech/substrate` to simplify upgrading to new releases.
```
Comment (cdamian) [inline docs/DEVELOPING.md:87]: Shouldn't this be the same command that we use in our CI - `cargo fmt -- --check`?

**Review by cdamian — APPROVED** [review]
LGTM, thank you for taking care of this @wischli!

File: docs/DEVELOPING.md [hunk 2]
```diff
@@ -172,9 +167,7 @@ The high level flow to upgrade to a newer version of Polkadot is:
     ```shell
     export POLKADOT_NEW_VERSION="<version>"; # for example, 0.9.32
 
-    diener update --polkadot --branch release-v$POLKADOT_NEW_VERSION;
-    diener update --substrate --branch polkadot-v$POLKADOT_NEW_VERSION;
-    diener update --cumulus --branch polkadot-v$POLKADOT_NEW_VERSION;
+    diener update --polkadot-sdk --branch release-v$POLKADOT_NEW_VERSION;
```
Comment (lemunozm) [inline docs/DEVELOPING.md:170]: Just notify this behavior has not been merged yet: https://github.com/paritytech/diener/pull/45

File: docs/DEVELOPING.md (hunk 1 above)
Comment (wischli) [reply docs/DEVELOPING.md:87]: Good point, I actually believe we should instead change the CI to check with `--all` 😅. IMO `fmt --all` here has no downsides as it is a superset of `fmt`. I also have to admit that I am not sure whether there is a difference between `cargo fmt` and `cargo fmt --all` for our workspace.

File: docs/DEVELOPING.md (hunk 2 above)
Comment (wischli) [reply docs/DEVELOPING.md:170]: Are you fine with keeping as proposed and iterating over this as part of the Polkadot SDK updates after v1.1.0?

Comment (lemunozm) [reply docs/DEVELOPING.md:170]: Yeah! Sure, I was only notifying

Comment (lemunozm) [reply docs/DEVELOPING.md:170]: Click the button 🚀

**Pull request merged**